microcosm_pubsub/codecs.py

```diff
@@ -79,9 +79,13 @@ def encode(self, dct: Optional[Dict[str, Any]] = None, **kwargs) -> str:
         message = dct.copy() if dct else dict()
         message.update(kwargs)
 
-        # Note that `dump` doesn't perform any validation
         dumped = self.schema.dump(message)
 
+        # NB: Schema dump doesn't run validation
+        errors = self.schema.validate(dumped)
+        if errors:
+            raise ValidationError(message=errors)
```

Comment:
Fields that are actually not in the schema will just be ignored by `dump` (since it doesn't validate), meaning that will be absent from `dumped`. So the only unknown fields at this point will be `dump_only` fields, that `validate` doesn't take into account in the schema.